src/transport/SessionManager.cpp

```diff
@@ -58,6 +58,31 @@ using Transport::SecureSession;
 namespace {
 Global<GroupPeerTable> gGroupPeerTable;
 
+/// RAII class for iterators that guarantees that Release() will be called
+/// on the underlying type
+template <typename Releasable>
+class AutoRelease
+{
+public:
+    AutoRelease(Releasable * iter) : mIter(iter) {}
+    ~AutoRelease() { Release(); }
+
+    Releasable * operator->() { return mIter; }
+    const Releasable * operator->() const { return mIter; }
+
+    bool IsNull() const { return mIter == nullptr; }
+
+    void Release()
+    {
+        VerifyOrReturn(mIter != nullptr);
+        mIter->Release();
+        mIter = nullptr;
+    }
+
+private:
+    Releasable * mIter;
```

Comment:
init with nullptr

Reply:
Updated, however existing constructor initialises this member, so I think the nullptr bit does not do anything. Probably useful only in case the class changes in the future in some way to allow a default constructor or if we add more constructors.